File: packages/ts/client/src/txbuilder/updateMarket.ts
import { Address, IInstruction, TransactionSigner } from "@solana/kit";

import {
  getMarketAddress,
  getTunaConfigAddress,
  getUpdateMarketInstruction,
  UpdateMarketInstructionDataArgs,
} from "../index.ts";

export async function updateMarketInstruction(
  authority: TransactionSigner,
  pool: Address,
  args: UpdateMarketInstructionDataArgs,
): Promise<IInstruction> {
  const tunaConfig = (await getTunaConfigAddress())[0];
  const marketAddress = (await getMarketAddress(pool))[0];

  return getUpdateMarketInstruction({
    authority,
    tunaConfig,
    market: marketAddress,
    ...args,
  });
}

File: rust-sdk/client/src/txbuilder/update_market.rs
use crate::instructions::{UpdateMarket, UpdateMarketInstructionArgs};
use crate::{get_market_address, get_tuna_config_address};
use solana_instruction::Instruction;
use solana_pubkey::Pubkey;

pub fn update_market_instruction(authority: &Pubkey, pool: &Pubkey, args: UpdateMarketInstructionArgs) -> Instruction {
    let tuna_config_address = get_tuna_config_address().0;
    let market_address = get_market_address(pool).0;

    let ix_builder = UpdateMarket {
        authority: *authority,
        tuna_config: tuna_config_address,
        market: market_address,
    };

    ix_builder.instruction(args)
}
